presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergPageSink.java

```diff
@@ -135,6 +165,37 @@ public IcebergPageSink(
         this.pagePartitioner = new PagePartitioner(pageIndexerFactory,
                 toPartitionColumns(inputColumns, partitionSpec),
                 session);
+        this.sortOrder = requireNonNull(sortOrder, "sortOrder is null");
+        this.sortedWritingEnabled = isSortedWritingEnabled(session);
+        this.sortingFileWriterBufferSize = requireNonNull(sortingFileWriterBufferSize, "sortingFileWriterBufferSize is null");
+        this.sortingFileWriterMaxOpenFiles = sortingFileWriterMaxOpenFiles;
+        String tempDirectoryPath = locationProvider.newDataLocation("sort-tmp-files");
+        this.tempDirectory = new Path(tempDirectoryPath);
+        this.typeManager = requireNonNull(typeManager, "typeManager is null");
+        this.pageSorter = requireNonNull(pageSorter, "pageSorter is null");
+        this.columnTypes = getColumns(outputSchema, partitionSpec, typeManager).stream()
```

Comment:
Make the field type ImmutableList if that's what you're storing here. Yes, the java Collections API is broken and has been since Java 1.1

Reply:
I usually prefer to use generic collection interfaces like collection, list rather than a concrete field type. So if needed, it can be changed from single place(at construction time)